test/end2end_test.go

```diff
@@ -31,7 +31,7 @@
  *
  */
 
-package grpc_test
+package test
```

Comment:
While you're here, would you mind moving this file up to the repo root and leaving it in `package grpc_test`? The current makeup of `package test` is confusing, where's it's half test utilities and half real tests.